server/src/repositories/media.repository.ts

```diff
@@ -45,7 +46,8 @@ export class MediaRepository implements IMediaRepository {
   }
 
   async generateThumbnail(input: string | Buffer, output: string, options: ThumbnailOptions): Promise<void> {
-    const pipeline = sharp(input, { failOn: 'error', limitInputPixels: false })
+    // some invalid images can still be processed by sharp, but we want to fail on them by default to avoid crashes
+    const pipeline = sharp(input, { failOn: PROCESS_INVALID_IMAGES ? 'none' : 'error', limitInputPixels: false })
```

Comment:
Not really sure if using an import from constants is the right choice here. Might also include it in the `options` object, but I guess that's something that shouldn't be configurable in the UI?

Reply:
For testing, I think it should either be a field in `options` or the env should be provided through a function like [here](https://github.com/immich-app/immich/blob/45cdd8a4b5cddc2af8360662a0e5d3f4d1e43b87/server/src/database.config.ts#L36).

Reply:
Done that and added a test case. Not too sure about accessing the env directly in the service, let me know if you have other suggestions :)